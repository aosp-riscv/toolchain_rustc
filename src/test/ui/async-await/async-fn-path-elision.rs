// edition:2018

<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
#![feature(async_await)]
#![allow(dead_code)]

=======
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
struct HasLifetime<'a>(&'a bool);

async fn error(lt: HasLifetime) { //~ ERROR implicit elided lifetime not allowed here
    if *lt.0 {}
}

fn no_error(lt: HasLifetime) {
    if *lt.0 {}
}

fn main() {}
