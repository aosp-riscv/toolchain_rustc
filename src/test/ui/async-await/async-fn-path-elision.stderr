error[E0726]: implicit elided lifetime not allowed here
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
  --> $DIR/async-fn-path-elision.rs:8:20
=======
  --> $DIR/async-fn-path-elision.rs:5:20
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
   |
LL | async fn error(lt: HasLifetime) {
   |                    ^^^^^^^^^^^- help: indicate the anonymous lifetime: `<'_>`

error: aborting due to previous error

